src/lib.rs

```diff
@@ -109,6 +109,38 @@ pub struct RewardRelationship<AccountId, Balance, BlockNumber, MintId, Recipient
     total_reward_missed: Balance,
 }
 
+impl<
+AccountId: Clone, 
+Balance: Clone, 
+BlockNumber: Clone, 
+MintId: Clone, 
+RecipientId: Clone
+> RewardRelationship<AccountId, Balance, BlockNumber, MintId, RecipientId> {
+
+    /// Verifies whether relationship is active
+    pub fn is_active(&self) -> bool {
+        self.next_payment_at_block.is_some()
+    }
+
+    /// Make clone which is activated.
+    pub fn clone_activated(&self, start_at: &BlockNumber) -> Self {
+
+        Self {
+            next_payment_at_block: Some((*start_at).clone()),
```

Comment:
If you add `Copy` trait bound for `BlockNumber` template argument, `start_at` can be passed by value instead of reference, and this can become cleaner: `Some(start_at)`

Reply:
cosmetic changes can wait.